src/api/spec/controllers/webui/project_controller_spec.rb

```diff
@@ -799,4 +799,129 @@
       it { expect(user.home_project.repositories.find_by(name: 'valid_name').repository_architectures.count).to eq(1) }
     end
   end
+
+  describe 'POST #save_distributions' do
+    let(:another_project_repository) { create(:repository, project: another_project) }
+    let(:target_repository) { create(:repository, project: openSUSE_project) }
+    let(:distribution) do
+      create(:distribution, reponame: another_project_repository.name,
+                            project: openSUSE_project, repository: target_repository.name, architectures: ['i586'])
+    end
+    let(:distribution_without_target_repo) { create(:distribution, reponame: another_project_repository.name, project: openSUSE_project) }
+    let(:create_distributions_same_repo) do
+      create_list(:distribution, 2, reponame: another_project_repository.name, project: openSUSE_project, repository: target_repository.name)
+    end
+    let(:create_distributions_other_reponame) { create_list(:distribution, 2, reponame: 'another_repon_name') }
+
+    before do
+      login user
+    end
+
+    context "without any distributions passed" do
+      before do
+        post :save_distributions, project: user.home_project
+      end
+
+      it { expect(flash[:success]).to eq("Successfully added repositories") }
+      it { is_expected.to redirect_to(action: :repositories, project: user.home_project) }
+      it { expect(assigns(:project).repositories.count).to eq(0) }
+    end
+
+    context "with a distribution but without target repository" do
+      it "Raises an ActiveRecord::RecordNotFound exception" do
+        expect do
+          post :save_distributions, project: user.home_project,
+                                    distributions: [distribution_without_target_repo.reponame]
+        end.to raise_error ActiveRecord::RecordNotFound
+      end
+    end
+
+    context "with two distributions using same repository" do
```

Comment:
Oh! You are right... and this is already tested at Repository spec (line 7). I will delete this, thanks